Backend/Tumultu/Application/Files/Commands/CreateFileCommand.cs

```diff
@@ -32,27 +28,38 @@ public async Task<Guid> Handle(CreateFileCommand request, CancellationToken canc
         IEnumerable<FileEntity> filesWithSameSignature = 
             await _repository.GetAllByAnySignature(md5, sha1, sha256);
 
+
+        FileEntity workingEntity;
+
         // this file already exists
-        if (filesWithSameSignature.Any())
+        if (filesWithSameSignature.Count() > 0)
```

Comment:
This might trigger 2 enumerations of the collection

Reply:
As that object is IEnumerable, we should probably switch to Any() and discard these change.